fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -756,20 +772,16 @@ public static FHIRSearchContext parseQueryParameters(Class<?> resourceType,
                           // Get the search parameter from our filtered set of applicable SPs for this resource type.
                           searchParameter = getSearchParameter(resType, parameterCode);
                           if (searchParameter == null) {
-                              String msg =
-                                      "Search parameter '" + parameterCode + "' for resource type '"
-                                              + resType + "' was not found.";
-                              throw SearchExceptionUtil.buildNewInvalidSearchException(msg);
+                              throw SearchExceptionUtil.buildNewInvalidSearchException(
+                                  String.format(SEARCH_PARAMETER_NOT_FOUND, parameterCode, resType));
```

Comment:
slight comment

In Exception cases String.format, we as a team deem it to be OK, but it is actually slower than assembling a string.  (I mention this as John and I had a deep discussion about StringBuilding performance when I made a similar change in this class).

Reply:
+1. And to add to this, although it's not the case here it's important exceptions are never used for flow control in the "happy path" case because they are expensive to build.